lib/ex_aws/request.ex

```diff
@@ -84,9 +86,11 @@ defmodule ExAws.Request do
     {:attempt, attempt + 1}
   end
 
-  # TODO: make exponential
-  # TODO: add jitter
   def backoff(attempt) do
-    :timer.sleep(attempt * 1000)
+    (@base_backoff_in_ms * :math.pow(2, attempt))
+    |> min(@max_backoff_in_ms)
+    |> trunc
+    |> :rand.uniform
```

Comment:
How do you feel about the possibility of me amending to the PR to make some of these values configurable at the service / request level (e.g. the base/max sleep time, retry count, and type of jitter)?

It looks like the aws java sdk allows some configuration of some things related to retries, and as an end user I could see wanting to handle retries differently in different situations. e.g. if I have a service with an API call that does an insert into SQS or Dynamo I may only want to retry 5 times over the course of 2-3 seconds before passing the backpressure amazon is giving me back to my client in the form of a 503 or something so they can engage their retry logic of my service, but in another scenarios, like some background process doing something with s3 files maybe i'm fine with retrying more times at potentially much longer intervals that would possibly be so long they'd cause something like nginx to declare a gateway timeout if the api service had those values.